dd-trace-core/src/main/java/datadog/trace/common/writer/WriterFactory.java

```diff
@@ -168,5 +154,32 @@ public static Writer createWriter(
     return remoteWriter;
   }
 
+  private static RemoteApi createDDIntakeRemoteApi(
+      Config config,
+      SharedCommunicationObjects commObjects,
+      DDAgentFeaturesDiscovery featuresDiscovery,
+      TrackType trackType) {
+    if (featuresDiscovery.supportsEvpProxy() && !config.isCiVisibilityAgentlessEnabled()) {
+      return DDEvpProxyApi.builder()
+          .agentUrl(commObjects.agentUrl)
+          .evpProxyEndpoint(featuresDiscovery.getEvpProxyEndpoint())
+          .trackType(trackType)
+          .build();
+
+    } else {
+      HttpUrl hostUrl = null;
+      if (config.getCiVisibilityAgentlessUrl() != null) {
+        hostUrl = HttpUrl.get(config.getCiVisibilityAgentlessUrl());
+        log.info(
+            "Using host URL '" + hostUrl + "' to report CI Visibility traces in Agentless mode.");
```

Comment:
```suggestion
            "Using host URL '{}' to report CI Visibility traces in Agentless mode.", hostUrl);
```

Prefer placeholders in log messages.